os/drivers/audio/alc5658.c

```diff
@@ -950,15 +949,19 @@ static int alc5658_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned lo
 
 		/* Report our preferred buffer size and quantity */
 
-#ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
 	case AUDIOIOC_GETBUFFERINFO: {
 		audvdbg("AUDIOIOC_GETBUFFERINFO:\n");
 		bufinfo = (FAR struct ap_buffer_info_s *)arg;
+#ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
 		bufinfo->buffer_size = CONFIG_ALC5658_BUFFER_SIZE;
 		bufinfo->nbuffers = CONFIG_ALC5658_NUM_BUFFERS;
+#else
+		bufinfo->buffer_size = priv->samprate / (1000 / CONFIG_ALC5658_I2S_TIMEOUT);
```

Comment:
Please add a comment indicating the formula used to calculate the buffer size. 
Also, would it be better to round off the buffer size to the next power of 2?

Reply:
Buffer size is proportional to sample rate & record/play back duration only. I could n't understand this formula.